### ISSUE TITLE
uses_systemd in cloudinit.distros doesn't properly detect systemd if VM isn't running

### ISSUE THREAD

**Issue by RussellAult** [canonical/cloud-init#5518]
# Bug report
https://github.com/canonical/cloud-init/blob/e0e6a427fdc6826bf7b11d52157a1c5f9b3dde4d/cloudinit/distros/__init__.py#L1608-L1613 returns a *false* `False` if the the target VM is being edited using offline tools (e.g. libguestfs), since `/run` is only populated on a running system

## Steps to reproduce the problem
`virt-customize -a <debian_cloud_image>.qcow2 --run-command 'cloud-init clean --machine-id'` should produce an `/etc/machine-id` of `undefined`, instead it deletes the file entirely (the non-systemd result)

## Environment details
- Cloud-init version: 24.2
- Operating System Distribution: Debian 12
- Cloud provider, platform or installer type: `virt-customize`

**Comment by holmanb** [comment]
This isn't a valid bug report.

**Comment by RussellAult** [comment]
> This isn't a valid bug report.

You're right! I accidentally hit enter while trying to type the title; sorry!

**Comment by RussellAult** [comment]
@holmanb Okay, can we try this again?
